### ISSUE TITLE
bug(steps): min_frequency in one_hot_ecode does not work

### ISSUE THREAD

**Issue by jitingxu1** [ibis-project/ibis-ml#146]
code to reproduce the error
```python
import ibis
import ibis_ml as ml

p = ibis.examples.penguins.fetch()
df_train, df_test = ml.train_test_split(p, unique_key="bill_length_mm", test_size=0.3, random_seed=12)
X_train = p.drop(["sex"])
y_train = p['sex'].cases((("male", 1), ("female", 0)), default=0).cast("int64").name("class")
recipe = ml.Recipe(
    ml.OneHotEncode(ml.string(), min_frequency=10),
)
recipe.fit(X_train, y_train)
```

error message
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[22], [line 8](vscode-notebook-cell:?execution_count=22&line=8)
      [4](vscode-notebook-cell:?execution_count=22&line=4) y_train = p['sex'].cases((("male", 1), ("female", 0)), default=0).cast("int64").name("class")
      [5](vscode-notebook-cell:?execution_count=22&line=5) recipe = ml.Recipe(
      [6](vscode-notebook-cell:?execution_count=22&line=6)     ml.OneHotEncode(ml.string(), min_frequency=10),
      [7](vscode-notebook-cell:?execution_count=22&line=7) )
----> [8](vscode-notebook-cell:?execution_count=22&line=8) recipe.fit(X_train, y_train)

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:562, in Recipe.fit(self, X, y)
    [547](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:547) """Fit a recipe.
    [548](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:548) 
    [549](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:549) Parameters
   (...)
    [559](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:559)     Returns the same instance.
    [560](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:560) """
    [561](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:561) table, targets, index = normalize_table(X, y)
--> [562](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:562) self._fit_table(table, targets, index)
    [563](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:563) return self

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:511, in Recipe._fit_table(self, table, targets, index)
    [508](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:508)     table = table.drop(index)
    [510](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:510) for step in self.steps:
--> [511](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:511)     step.fit_table(table, metadata)
    [512](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:512)     table = step.transform_table(table)
    [514](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/core.py:514) self.metadata_ = metadata

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:150, in OneHotEncode.fit_table(self, table, metadata)
    [146](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:146)     else:
    [147](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:147)         to_compute.append(column)
    [149](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:149) categories.update(
--> [150](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:150)     _compute_categories(
    [151](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:151)         table, to_compute, self.min_frequency, self.max_categories
    [152](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:152)     )
    [153](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:153) )
    [155](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:155) self.categories_ = categories

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:67, in _compute_categories(table, columns, min_frequency, max_categories)
     [64](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:64)         return df["value"].sort_values().to_list()
     [66](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:66) for group_cols in groups.values():
---> [67](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:67)     query = ibis.union(*(collect(col) for col in group_cols))
     [68](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:68)     result_groups = query.execute().groupby("column")
     [70](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:70)     for col in group_cols:

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:67, in <genexpr>(.0)
     [64](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:64)         return df["value"].sort_values().to_list()
     [66](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:66) for group_cols in groups.values():
---> [67](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:67)     query = ibis.union(*(collect(col) for col in group_cols))
     [68](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:68)     result_groups = query.execute().groupby("column")
     [70](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:70)     for col in group_cols:

File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:44, in _compute_categories.<locals>.collect(col)
     [40](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:40) def collect(col: str) -> ir.Table:
     [41](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:41)     query = (
     [42](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:42)         table.select(value=col)
     [43](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:43)         .group_by("value")
---> [44](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:44)         .count("count")
     [45](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:45)         .mutate(column=ibis.literal(col))
     [46](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:46)     )
     [47](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_encode.py:47)     return query if max_categories is None else query.limit(max_categories)
```

**Comment by jitingxu1** [comment]
```python
query = (
                table.select(value=col)
                .group_by("value")
                .count("count")
                .mutate(column=ibis.literal(col))
            )
```
count() does not accept the extral argument.

**Comment by jitingxu1** [comment]
```
max_categories
        A maximum number of categories to include. If set, only the most
        frequent ``max_categories`` categories are kept.
```
the most frequent category is not respected, will resolve this too.

**Comment by deepyaman** [comment]
Resolved in #156
